### DISCUSSION TITLE
Enforcing collision between agents

### DISCUSSION THREAD

**Discussion by dreginato** [gama-platform/gama#45] [Q&A] [answered]
Hi everybody!
I am currently working on a model for decentralized networking on a plane; I have a population of agents who move according to some simple rules involving the concentration of "pheromones" left by other agents and the presence of food sources.
I would like the agents to move while also taking into account the fact that they have a geometry, so that they do not overlap and I can avoid pile-ups on the food sources.
I cannot think of a way to implement this, and I found nothing on the documentation.
Do you think it can be done? I'd really appreciate any help. Thanks!

**Comment by lesquoyb** [comment]
I'm sure @ptaillandier has some great insights on this

**Reply by chapuisk** [reply, marked as answer]
@dreginato Did you get an answer for this? If not, there is one skill that do exactly what you are looking for: pedestrian (look at the plugin models for examples).

**Reply by dreginato** [reply]
Yes! I eventually found out about the "pedestrian" skill on my own, but thank you anyway!
